utility_scripts/quota_audit.py

```diff
@@ -0,0 +1,171 @@
+auth_url = os.environ.get('OS_AUTH_URL')
+
+"""
+# Uncomment this section to authenticate with hard coded values
+#admin_user = 'admin'
+#admin_pwd = 'secret'
+#admin_project = 'admin'
+#auth_url = 'http://some-auth-url'
+"""
+
+auth = v2.Password(auth_url=auth_url,
+                   username=admin_user,
+                   password=admin_pwd,
+                   tenant_name=admin_project)
+
+sess = session.Session(auth=auth)
+keystone = client.Client(session=sess)
+ks_projects = keystone.projects.list()
+neutron = nclient.Client(session=sess)
+nova = novaclient.Client(2, session=sess)
+cinder = cinderclient.Client(session=sess)
+
+moc_standards = {'subnet': 10, 'router': 10, 'port': 10, 'network': 5,
+                 'floatingip': 2, 'security_group': -1,
+                 'security_group_rule': -1, 'ram': 51200, 'gigabytes': 1000,
+                 'snapshots': 10, 'volumes': 10,
+                 'injected_file_content_bytes': 10240, 'injected_files': 5,
+                 'metadata_items': 128, 'instances': 10, 'cores': 20}
+
+
+def diff_moc_quotas(project_quotas):
+
+    different_quotas = {}
+    for each_quota in moc_standards:
+        if moc_standards[each_quota] == project_quotas[each_quota]:
+            continue
+        else:
+            different_quotas[each_quota] = project_quotas[each_quota]
+    return different_quotas
+
+
+def to_single_dict(no_q, ci_q, ne_q):
+
+    combined = {}
+    combined.update(no_q)
+    combined.update(ci_q)
+    combined.update(ne_q)
+    return combined
+
+
+def proj_to_request_dict(project_name):
+ 
+    request_dict = {}
+    with open("Real.csv", "rb") as source:  # change file name
+        reader = csv.DictReader(source)
+        for row in reader:
+            if row['OpenStack project name'] == project_name:
+                request_dict['instances'] = row['Instances']
+                request_dict['cores'] = row['VCPUs']
+                request_dict['ram'] = row['RAM']
+                request_dict['floatingip'] = row['Floating IPs']
+                request_dict['network'] = row['Networks']
+                request_dict['port'] = row['Ports']
+                request_dict['volumes'] = row['Volumes']
+                request_dict['snapshots'] = row['Snapshots']
+                request_dict['gigabytes'] = row['Volume & Snapshot Storage']
+    return request_dict
+
+
+def isolate_requests(dict_with_blanks):  # singles out the requests
+
+    configured_dict = dict((k, v) for k, v in dict_with_blanks.items() if v)
+    for k in configured_dict:
+        if k == 'ram':
+            configured_dict[k] = str(int(configured_dict[k]) * 1024)
+    return configured_dict
+
+
+def compare_request_with_real(requested_quotas, all_quotas):
+    
+    for key in requested_quotas:
+        if int(requested_quotas[key]) == all_quotas[key]:
+            continue
+        else:
+            return False
+    return True
+
+
+all_neutron_quotas = [q for q in neutron.list_quotas()['quotas']]
```

Comment:
I don't think so. `q for q` is just a passthrough, and at least for me `type(neutron.list_quotas()['quotas'])` is already list